bin/tictactoe

```diff
@@ -1,3 +1,71 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+#  ./bin/tictactoe
+
+  def choose_players
+    puts "     How many soon-to-be-enthralled players do we have today?"
+    puts "     Please enter 1 (for 1-player), 2 (for 2-players) or 0 (for computer-only mode)"
+      players = gets.strip
+      if players == "0"
+          player1 = Computer.new("X")
+          player2 = Computer.new("O")
+          game = Game.new(player1, player2)
+          game.play
+        elsif players == "1"
+          who_goes_first?
+        elsif players == "2"
+          Game.new.play
+          else
+            puts "Invalid entry. Please only enter 1 (for 1-player), 2 (for 2-players) or 0 (for computer-only mode)"
+            end
+      play_again?
+  end
+
+
+    def who_goes_first?
+      puts "     Who would you like to go first?"
+      puts "     Type M if you would like to go first, or type C if you would like the computer to play first"
+      first = gets.chomp.downcase
+        if first == "m"
+          player1 = Human.new("X")
+          player2 = Computer.new("O")
+          game = Game.new(player1, player2)
+          game.play
+        elsif first == "c"
+          puts "AI selected"
+        else
+          puts " "
+          puts "     Invalid entry. Please only enter M or C."
+          who_goes_first?
+      end
+    end
+
+  def play_again?
+    puts " "
+    puts "     Would you like to play again?"
+    puts "     Type Y if you would like to continue, or N if you would like to stop"
+    desire = gets.strip.downcase
+      if desire == "y"
+        choose_players
+      elsif desire == "n"
+        puts " "
+        puts "     Thanks for playing. Drive safely and see you next time!"
+      else
+        puts " "
+        puts "     Invalid entry. Please only enter Y or N."
+          play_again?
+      end
+  end
+
+
+puts " "
+puts "     *****Welcome to Tic Tac Toe!*****"
+puts "     The GREATEST and most ENJOYALBE game of all-time!"
+puts " "
+choose_players
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.